api/controllers/StatsController.js

```diff
@@ -6,6 +6,40 @@ dayjs.extend(isBetween);
 /////////////
 // Helpers //
 /////////////
+/**
+ * Find the season in which a given timestamp occurred using binary search
+ * @param {*} sortedSeasons - Seasons sorted (OLDEST FIRST)
+ * @param {*} timeStamp - Desired time in millis since epoch
+ * @returns the season | null if none is found
+ */
+function getSeasonFromTimeStamp(sortedSeasons, timeStamp) {
+  let currentIndex;
+  let minIndex = 0;
+  let maxIndex = sortedSeasons.length - 1;
+  // Sanity check in case list is unsorted or start & endTimes are out of whack
+  let numTries = 0;
+  while (numTries < sortedSeasons.length) {
+    currentIndex = Math.floor((minIndex + maxIndex) / 2);
```

Comment:
Is this just getting the middle result? Is there not a better way to get the current season? I might also be reading this wrong, PRing on my phone from vacation. 😎

Reply:
Yeah it just calculates true middle index between `minIndex` and `maxIndex`. I can't think of a simpler way than averaging and using `Math.floor()` to truncate to int, but I'm open to suggestions